src/index.ts

```diff
@@ -72,7 +72,21 @@ export const temporal = (<TState>(
         // The order of the get() and set() calls is important here.
         const pastState = options?.partialize?.(get()) || get();
         set(...args);
-        curriedHandleSet(pastState);
+        const currentState = options?.partialize?.(get()) || get();
```

Comment:
For consistency, could you also make sure this change takes place in the `setState` function on line 58? Thanks!

Reply:
Yeah can do. To be honest I haven't quite been able to wrap my head around what is going on on line 60. You happen to have a quick way of explaining what is going on there? 😅

Reply:
Sure! The quick explanation is that zustand has two ways of setting state; one is used often, and it's during the creation method where `set` is used within the callback and the second is after the store is created, calling `myStore.setState` (which is less used):

```ts
const myStore = create<MyState>()((set, get) => ({ increment: set((state) => ({ bears: state.bears + 1)});

myStore.setState({ bears: 99 });
```

The `zundo` user might prefer one setter over the other, but to make sure the middleware acts the same, we need to modify both setters. This was really brief, so I'm happy to elaborate more.

Reply:
The `set` version is updated to track history within the `config` function call, and the `setState` version is directly mutated on the user's `store` object (the object returned by `zustand/create` or `zustand/createStore`.

Reply:
Beautiful, thank you!